caikit/runtime/client/remote_model_initializer.py

```diff
@@ -0,0 +1,92 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+The RemoteModelInitializer loads a RemoteModuleConfig as an empty Module that
+sends all requests to an external runtime server
+
+Configuration for RemoteModelInitializer lives under the config as follows:
+
+model_management:
+    initializers:
+        <initializer name>:
+            type: REMOTE
+"""
+# Standard
+from typing import Optional, Type
+
+# First Party
+import aconfig
+import alog
+
+# Local
+from caikit.core.exceptions import error_handler
+from caikit.core.model_management.factories import model_initializer_factory
+from caikit.core.model_management.model_initializer_base import ModelInitializerBase
+from caikit.core.modules import ModuleBase
+from caikit.runtime.client.remote_config import RemoteModuleConfig
+from caikit.runtime.client.remote_module_base import construct_remote_module_class
+
+log = alog.use_channel("RINIT")
+error = error_handler.get(log)
+
+
+class RemoteModelInitializer(ModelInitializerBase):
+    __doc__ = __doc__
+    name = "REMOTE"
+
+    def __init__(self, config: aconfig.Config, instance_name: str):
+        """Construct with the config"""
+        self._instance_name = instance_name
+        self._module_class_map = {}
+
+    def init(self, model_config: RemoteModuleConfig, **kwargs) -> Optional[ModuleBase]:
+        """Given a RemoteModuleConfig, initialize a RemoteModule instance"""
+
+        # Ensure the module config was produced by a RemoteModelFinder
+        error.type_check(
```

Comment:
Same comment here about error semantics. It is pretty nice to just have this kind of `error.type_check` here though

Reply:
I think I like your original comment about error semantics. Even though the type_check is pretty I think consistency would be clearer. I'll update this to a log message and return